### ISSUE TITLE
Renaming multiple columns fails on SQL Server

### ISSUE THREAD

**Issue by jcfisher** [tidyverse/dplyr#3228]
On SQL Server, renaming multiple columns fails with the error:

```
Error in names(select)[match(old_vars, vars)] <- new_vars : 
  NAs are not allowed in subscripted assignments
```

Renaming single columns works.  This seems similar to [an old, closed issue (#2860)](https://github.com/tidyverse/dplyr/issues/2860).

A minimal reprex follows:

```r
library(tidyverse)
library(dbplyr)
con <- DBI::dbConnect(
  odbc::odbc(),
  driver = "{SQL Server}",
  database = "AdventureWorks2012",
  uid = "sqlfamily",
  pwd = "sqlf@m1ly",
  server = "mhknbn2kdz.database.windows.net",
  port = 1433
)

dept <- con %>% 
  tbl(in_schema("HumanResources", "Department"))
    
# Fails
dept %>% 
  rename(test1 = DepartmentID, test2 = Name)
# Error in names(select)[match(old_vars, vars)] <- new_vars : 
#   NAs are not allowed in subscripted assignments

# Also fails
dept %>% 
  rename_at(.vars = vars(contains("Name")), .funs = funs(paste0(., "_foo")))
# Error in names(select)[match(old_vars, vars)] <- new_vars : 
#   NAs are not allowed in subscripted assignments
```
Thanks for your consideration on this, and please let me know if there's anything I can do to help.

**Comment by krlmlr** [comment]
Thanks. I've seen a few recent issues concerning `mutate()` problems on database backends, could this be a duplicate?

**Comment by jiho** [comment]
This is also true for SQlite

```r
d <- data.frame(a=1:3, b=4:6)
f <- src_sqlite("test.sqlite3", create=TRUE)
copy_to(dest=f, df=d)
d_db <- tbl(f, "d")

rename(d   , x=a)
# -> works

rename(d   , x=a, y=b)
# -> works

rename(d_db, x=a)
# -> works

rename(d_db, x=a, y=b)
# -> fails

# Error in names(select)[match(old_vars, vars)] <- new_vars : 
# NAs are not allowed in subscripted assignments
```

and PostgreSQL (no example but the message is the same, so I assume the error is the same too).

**Comment by jiho** [comment]
Sorry, seems to have been fixed cf #2943

**Comment by krlmlr** [comment]
Confirmed that SQLite reprex works now. You can use `memdb_frame()` for shorter reprexes.
